tutorials/CHT/flow-over-plate/buoyantPimpleFoam-laplacianFoam_nearest-projection/README.md

```diff
@@ -0,0 +1,84 @@
+
+### Changes in `precice-config.xml`
+In order to map from face nodes to face centers, both meshes need to be specified. The nodes-based mesh uses the write data and the centers-based mesh uses the read data. Have a look in the given `precice-config.xml` in this tutorial. Example: `Temperature` is calculated by the `Fluid` participant and passed to the `Solid` participant. Therefore, it is the write data of the participant `Fluid` and the read data of the participant `Solid`. This results in the following two meshes for this data:
+```
+<mesh name="Fluid-Mesh-Nodes">
+  <use-data name="Temperature"/>
+</mesh>
+<mesh name="Solid-Mesh-Centers">
+  <use-data name="Temperature"/>
+</mesh>
+```
+All further changes follow from this interface splitting. Have a look in the given config files for all details.
+
+### Changes in `precice-adapter-config.yml`
+
+By default, the OpenFOAM adapter doesn't provide any connectivity information. Therefore, a new boolean variable called `provideMeshConnectivity` is introduced. This variable is associated to each interface and can be set accordingly. Note: Mesh connectivity can only be provided in case `locationsType: faceNodes` is set (see section Adapter Implementation). Similar to the interface splitting in the `precice-config.xml` file, the interfaces also need to be defined in the `yml` file. For example:
+
+```
+interfaces:
+- mesh: Fluid-Mesh-Centers
+  locations: faceCenters         # default
+  provideMeshConnectivity: false # default
+  patches:
+  - interface
+  read-data: Heat-Flux
+- mesh: Fluid-Mesh-Nodes
+  locations: faceNodes
+  provideMeshConnectivity: true
+  patches:
+  - interface
+  write-data: Temperature
+```
+The participant `Fluid` has its read data `Heat-Flux`, which is read on the `faceCenters`, and its write data `Temperature`, which is written to the `faceNodes`. The mesh connectivity needs only to be provided in case of the `faceNodes`, using the option `provideMeshConnectivity: true`.
+
+## General Notes
+
+Since you now define mesh connectivity on your interface, you can export your coupling interface with the tag `<export:vtk directory="preCICE-output" />` in your `precice-config.xml`. Make sure you have created the specified target directory (preCICE-output in the example files) in your local simulation directory, otherwise preCICE will complain about it.
+
+Visualizing these files (e.g. on ParaView) will show a triangular mesh, even though you use hexahedral meshes. This has nothing to do with your mesh and is just caused by the way the connectivity is defined in preCICE. As described above, the function `setMeshTriangles` is used to define the connectivity. Hence, every interface cell/face is represented by two triangles. The following image should give you an impression of a possible triangulated coupling mesh, which consists purely of hexahedral cells:
```

Comment:
Could you please describe how to get this image on ParaView? I am trying to reproduce it, but applying a Delaunay2D for example does not give anything interpretable.

Reply:
Are you sure you selected the right interface mesh? Only the 'Node' mesh provide connectivity. You don't need to apply special features. Just select 'Surface with Edges' in the user interface.

Reply:
This is a useful comment, which should go into the README.

However, this was not my issue, as shown here: https://github.com/precice/precice/issues/368#issuecomment-487040263